lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/AzureBatchBootstrapREEFLauncher.java

```diff
@@ -0,0 +1,123 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.bridge.client;
+
+import org.apache.avro.io.DecoderFactory;
+import org.apache.avro.io.JsonDecoder;
+import org.apache.avro.specific.SpecificDatumReader;
+import org.apache.reef.annotations.audience.Interop;
+import org.apache.reef.reef.bridge.client.avro.AvroAzureBatchJobSubmissionParameters;
+import org.apache.reef.runtime.azbatch.client.AzureBatchRuntimeConfiguration;
+import org.apache.reef.runtime.azbatch.client.AzureBatchRuntimeConfigurationCreator;
+import org.apache.reef.runtime.common.REEFLauncher;
+import org.apache.reef.tang.Configuration;
+import org.apache.reef.tang.Tang;
+import org.apache.reef.tang.exceptions.InjectionException;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This is a bootstrap launcher for YARN for submission from C#. It allows for Java Driver
+ * configuration generation directly on the Driver without need of Java dependency if REST
+ * submission is used. Note that the name of the class must contain "REEFLauncher" for the time
+ * being in order for the Interop code to discover the class.
+ */
+@Interop(CppFiles = "DriverLauncher.cpp")
+public final class AzureBatchBootstrapREEFLauncher {
+  private static final Logger LOG = Logger.getLogger(AzureBatchBootstrapREEFLauncher.class.getName());
+
+  public static void main(final String[] args) throws IOException, InjectionException {
+    LOG.log(Level.INFO, "Entering BootstrapLauncher.main().");
+
+    if (args.length != 1) {
+      final StringBuilder sb = new StringBuilder();
+      sb.append("[ ");
+      for (String arg : args) {
+        sb.append(arg);
+        sb.append(" ");
+      }
+
+      sb.append("]");
+
+      final String message = "Bootstrap launcher should have one configuration file input, specifying the" +
+          "job submission parameters to be deserialized to create the YarnDriverConfiguration on the fly." +
+          " Current args are " + sb.toString();
+
+      throw fatal(message, new IllegalArgumentException(message));
+    }
+
+    LOG.log(Level.INFO, "Entering BootstrapLauncher.main(). {0}", args[0]);
+
+    try {
+
+      final File partialConfigFile = new File(args[0]);
+      final AzureBatchBootstrapDriverConfigGenerator azureBatchBootstrapDriverConfigGenerator =
+          Tang.Factory.getTang().newInjector(generateConfigurationFromJobSubmissionParameters(partialConfigFile))
+              .getInstance(AzureBatchBootstrapDriverConfigGenerator.class);
+      REEFLauncher.main(new String[]{azureBatchBootstrapDriverConfigGenerator.writeDriverConfigurationFile(args[0])});
```

Comment:
You should not be writing to the same file you received as an argument. All inputs should remain static. Please use a temp file.

Reply:
Seems some misunderstanding here : args[0] is the json file contains as input. In line 77, it will read the data from job-submission-params.json file and write the configuration into a different file "driver.conf"